awscli/customizations/s3/filegenerator.py

```diff
@@ -208,6 +214,12 @@ def list_files(self, path, dir_op):
                             yield x
                     else:
                         size, last_update = get_file_stat(file_path)
+                        if last_update is EPOCH_TIME:
```

Comment:
I am fine with this approach but can we abstract this logic to a new method so we do not have to copy the logic twice?